lib-impl/src/main/java/org/ergoplatform/appkit/impl/ScalaBridge.scala

```diff
@@ -170,33 +171,34 @@ object ScalaBridge {
     }
   }
 
-//  implicit val isoExplTransactionOutput: Iso[TransactionOutput, ErgoBox] = new Iso[TransactionOutput, ErgoBox] {
-//    override def to(boxData: TransactionOutput): ErgoBox = {
-//      val tree = boxData.getErgoTree.convertTo[ErgoTree]
-//      val tokens = boxData.getAssets.convertTo[Coll[(TokenId, Long)]]
-//      val regs = boxData.getAdditionalRegisters().convertTo[AdditionalRegisters]
-//      new ErgoBox(boxData.getValue, tree,
-//        tokens, regs,
-//        ModifierId @@ boxData.getTransactionId,
-//        boxData.getIndex.shortValue,
-//        boxData.getCreationHeight)
-//    }
-//
-//    override def from(box: ErgoBox): ErgoTransactionOutput = {
-//      val assets = box.additionalTokens.convertTo[List[Asset]]
-//      val regs = isoRegistersToMap.from(box.additionalRegisters)
-//      val out = new ErgoTransactionOutput()
-//          .boxId(ErgoAlgos.encode(box.id))
-//          .value(box.value)
-//          .ergoTree(ErgoAlgos.encode(TreeSerializer.serializeErgoTree(box.ergoTree)))
-//          .assets(assets)
-//          .additionalRegisters(regs)
-//          .creationHeight(box.creationHeight)
-//          .transactionId(box.transactionId)
-//          .index(box.index)
-//      out
-//    }
-//  }
+  implicit val isoExplTransactionOutput: Iso[OutputInfo, ErgoBox] = new Iso[OutputInfo, ErgoBox] {
+    override def to(boxData: OutputInfo): ErgoBox = {
+      val tree = boxData.getErgoTree.convertTo[ErgoTree]
+      val tokens = boxData.getAssets.convertTo[IndexedSeq[AssetInstanceInfo]].sortBy(_.getIndex)
+        .map(asset => new ErgoToken(asset.getTokenId, asset.getAmount)).convertTo[JList[ErgoToken]].convertTo[Coll[(TokenId, Long)]]
+      val regs = boxData.getAdditionalRegisters.convertTo[AdditionalRegisters]
+      new ErgoBox(boxData.getValue, tree,
+        tokens, regs,
+        ModifierId @@ boxData.getTransactionId,
+        boxData.getIndex.shortValue,
+        boxData.getCreationHeight)
+    }
+
+    override def from(box: ErgoBox): OutputInfo = {
+      val assets = box.additionalTokens.convertTo[List[Asset]]
+      val regs = isoExplRegistersToMap.from(box.additionalRegisters)
+      val out = new OutputInfo()
+          .boxId(ErgoAlgos.encode(box.id))
+          .value(box.value)
+          .ergoTree(ErgoAlgos.encode(TreeSerializer.serializeErgoTree(box.ergoTree)))
+          .assets(assets.map(asset => new AssetInstanceInfo().tokenId(asset.getTokenId).amount(asset.getAmount)))
+          .additionalRegisters(regs)
+          .creationHeight(box.creationHeight)
+          .transactionId(box.transactionId)
+          .index(box.index)
+      out
+    }
+  }
```

Comment:
Does the `to` and `from` mutually reverse. They are maybe not because of that `sortBy`.

Reply:
Good catch, the index needs to be set

Reply:
Done